src/Forms/MoneyField.php

```diff
@@ -175,8 +184,16 @@ public function setValue($value, $data = null)
             throw new InvalidArgumentException("Invalid currency format");
         }
 
+        // Check that currency is allowed, if not, swap out currency field for a text field so
+        // that user can alter the value as it will fail validation
+        $allowedCurrencies = $this->getAllowedCurrencies() ?? [];
+        $currency = $value['Currency'];
+        if ($currency && count($allowedCurrencies) && !in_array($currency, $allowedCurrencies)) {
+            $this->fieldCurrency = $this->buildCurrencyField(true);
+        }
```

Comment:
This provides the option to change the currency, but the validation error message is simply "Currency USD is not in the list of allowed currencies"

Either the validation message, or `setDescription()` on the currency field, needs to include the list of allowed currencies. Otherwise content authors will be going through the full list of ISO currencies trying to set it to the only one that's actually allowed.

Reply:
Updated